console/console-init/ui/src/Components/AddressSpace/AddressListFilter.tsx

```diff
@@ -0,0 +1,107 @@
+import * as React from "react";
+import {
+  DropdownMenu,
+  IDropdownOption
+} from "../Common/DropdownComponent";
+import {
+  InputGroup,
+  Button,
+  TextInput,
+  ButtonVariant
+} from "@patternfly/react-core";
+import { SearchIcon } from "@patternfly/react-icons";
+
+export interface IAddressListFilterProps {
+  onSearch: () => void;
+  onFilterSelect: (item: any) => void;
+  filterValue: string;
+  isTypeDropdownOpen: boolean;
+  onTypeSelect: (item: any) => void;
+  typeValue: string;
+  setTypeOpen: () => void;
+  isStatusDropdownOpen: boolean;
+  onStatusSelect: (item: any) => void;
+  statusValue: string;
+  setStatusOpen: () => void;
+}
+export const filterOptions: IDropdownOption[] = [
+    { value: "name", label: "Name" },
+    { value: "type", label: "Type" },
+    { value: "status", label: "Status" }
+  ],
+  typeOptions: IDropdownOption[] = [
+    { value: "queue", label: "Queue" },
+    { value: "topic", label: "Topic" },
+    { value: "subscripition", label: "Subscripition" },
+    { value: "mulitcast", label: "Mulitcast" },
+    { value: "anycast", label: "Anycast" }
+  ],
+  statusOptions: IDropdownOption[] = [
+    { value: "active", label: "Active" },
+    { value: "configuring", label: "Configuring" },
+    { value: "failed", label: "Failed" }
+  ];
+
+export const AddressListFilter: React.FunctionComponent<
+  IAddressListFilterProps
+> = ({
+  onSearch,
+  onFilterSelect,
+  filterValue,
+  isTypeDropdownOpen,
+  onTypeSelect,
+  typeValue,
+  setTypeOpen,
+  isStatusDropdownOpen,
+  onStatusSelect,
+  statusValue,
+  setStatusOpen
+}) => {
+  const [isFilterDropdownOpen, setIsFilterDropdwonOpen] = React.useState(false);
+  return (
+    <InputGroup>
+      <DropdownMenu
+        isOpen={isFilterDropdownOpen}
+        value={filterValue}
+        onSelect={onFilterSelect && setIsFilterDropdwonOpen}
+        setIsOpen={() => setIsFilterDropdwonOpen(!isFilterDropdownOpen)}
```

Comment:
Why should the `AddressListFilter` need to deal with the opened/closed state for this dropdown?